lib/ansible/modules/database/mongodb/mongodb_role.py

```diff
@@ -0,0 +1,452 @@
+
+
+def role_add(module, client, db_name, role, privileges, roles, authentication_restrictions, authentication_restrictions_supported):
+    db = client[db_name]
+
+    if authentication_restrictions_supported:
+        result = db.command('createRole',
+                            role,
+                            privileges=privileges,
+                            roles=roles,
+                            authenticationRestrictions=authentication_restrictions
+                            )
+    else:
+        result = db.command('createRole',
+                            role,
+                            privileges=privileges,
+                            roles=roles)
+
+    if result['ok'] != 1:
+        module.fail_json(msg='createRole failed', exception=traceback.format_exc())
+
+
+def role_update(module, client, db_name, role, privileges, roles, authentication_restrictions, authentication_restrictions_supported):
+    db = client[db_name]
+
+    if authentication_restrictions_supported:
+        result = db.command('updateRole',
+                            role,
+                            privileges=privileges,
+                            roles=roles,
+                            authenticationRestrictions=authentication_restrictions
+                            )
+    else:
+        result = db.command('updateRole',
+                            role,
+                            privileges=privileges,
+                            roles=roles
+                            )
+
+    if result['ok'] != 1:
+        module.fail_json(msg='updateRole failed', exception=traceback.format_exc())
+
+
+def role_compare(privileges, roles, authentication_restrictions, current, authentication_restrictions_supported):
+    if authentication_restrictions_supported:
+        # yes, rolesInfo does return single element list of lists
+        if current['authenticationRestrictions']:
+            current_restrictions = current['authenticationRestrictions'][0]
+        else:
+            current_restrictions = []
+
+        return sorted(current['roles']) == sorted(roles) \
+            and sorted(current['privileges']) == sorted(privileges) \
+            and sorted(current_restrictions) == sorted(authentication_restrictions)
+    else:
+        return sorted(current['roles']) == sorted(roles) \
+            and sorted(current['privileges']) == sorted(privileges)
+
+
+def role_remove(module, client, db_name, role):
+    db = client[db_name]
+    result = db.command('dropRole', role)
+
+    if result['ok'] != 1:
+        module.fail_json(msg='dropRole failed', exception=traceback.format_exc())
+
+
+# =========================================
+# Module execution.
+#
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            login_user=dict(default=None),
+            login_password=dict(default=None, no_log=True),
+            login_host=dict(default='localhost'),
+            login_port=dict(default='27017'),
```

Comment:
use `type='int'` here and you won't have to do manual conversion or checking manually